frontend/mux.go

```diff
@@ -586,25 +586,52 @@ func WithTargetForwardingHandler(ctx context.Context, client gwclient.Client, m
 // These are only added if the target is in the spec OR the spec has no explicit targets.
 func WithBuiltinHandler(key string, bf gwclient.BuildFunc) func(context.Context, gwclient.Client, *BuildMux) error {
 	return func(ctx context.Context, client gwclient.Client, m *BuildMux) error {
-		spec, err := m.loadSpec(ctx, client)
-		if err != nil {
-			return err
+		if !shouldLoadTarget(ctx, client, m, key) {
+			return nil
 		}
+		m.Add(key, bf, nil)
+		return nil
+	}
+}
 
-		if len(spec.Targets) > 0 {
-			t, ok := spec.Targets[key]
-			if !ok {
-				bklog.G(ctx).WithField("spec targets", maps.Keys(spec.Targets)).WithField("targetKey", key).Info("Target not in the spec, skipping")
-				return nil
-			}
+// shouldLoadTarget is used to determine if the spec is overriding the built-in
+// target with the same targetKey.
```

Comment:
Can you add a comment here to clarify the logic and rationale a bit more, i.e., we will always load a target if there are no targets specified in the spec, otherwise, we will only load the targets specified in the spec?

Reply:
Done.